qiskit_experiments/database_service/db_analysis_result.py

```diff
@@ -205,6 +205,8 @@ def _from_service_data(cls, service_data: Dict) -> "DbAnalysisResultV1":
         result_data = service_data.pop("result_data")
         value = result_data.pop("_value")
         extra = result_data.pop("_extra", {})
+        if extra:
+            chisq = extra.pop("reduced_chisq", None)
```

Comment:
The extra field is not what is saved in the chisq of the analysis result, this is just leftover metadata from certain curve fitting experiments. It should just be`result_data.pop("chisq", None)` here, but since we apply the `display_format` function for modifying chisq for UI display we will need to add a line to save the original without modification.

I think the correct thing to do here is update line 155 of the `save` function to store the non-modified chisq:

```python
result_data = {
            "_value": value,
            "_chisq": self._chisq,
            "_extra": self.extra,
            "_source": self._source,
        }
```
then this line should be
```suggestion
        chisq = result_data.pop("_chisq", None)
```

Reply:
I did as you suggest and it's working. But I think the correct thing would be (in another PR) to add `chisq` as a parameter in `create_analysis_result` and `update_analysis_result`. Is there a reason for `chisq` to be different from other attributes, like `quality` and `verified`?